fabric-item-api-v1/src/main/java/net/fabricmc/fabric/mixin/item/CrossbowItemMixin.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019, 2020 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.fabricmc.fabric.mixin.item;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+import net.fabricmc.fabric.api.item.v1.FabricCrossbowHooks;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.projectile.PersistentProjectileEntity;
+import net.minecraft.item.ArrowItem;
+import net.minecraft.item.CrossbowItem;
+import net.minecraft.item.ItemStack;
+import net.minecraft.world.World;
+
+@Mixin(CrossbowItem.class)
+public class CrossbowItemMixin {
+
+	@Inject(method = "createArrow", at = @At(value = "RETURN"), locals = LocalCapture.CAPTURE_FAILSOFT)
+	private static void createArrow(World world, LivingEntity entity, ItemStack crossbow, ItemStack arrow,
+			CallbackInfoReturnable<PersistentProjectileEntity> cir, ArrowItem arrowItem,
+			PersistentProjectileEntity persistentProjectileEntity) {
+
+		if (!(crossbow.getItem() instanceof CrossbowItem)) {
+			return;
+		}
+		((FabricCrossbowHooks) crossbow.getItem()).createArrow(arrowItem, persistentProjectileEntity);
+	}
+
+	@Inject(method = "getSpeed", at = @At(value = "HEAD"))
+	private static void tickMovement(ItemStack itemStack, CallbackInfoReturnable<Float> cir) {

```

Comment:
`tickMovement`? This I assume gets the speed at which the crossbow is charged at.